sanic/server.py

```diff
@@ -122,44 +126,26 @@ def on_message_complete(self):
 
     def write_response(self, response):
         try:
-            keep_alive = self.parser.should_keep_alive() \
-                            and not self.signal.stopped
-            self.transport.write(
+            keep_alive = STATUS.running and self.parser.should_keep_alive()
+            self.write(
                 response.output(
-                    self.request.version, keep_alive, self.request_timeout))
-            if not keep_alive:
-                self.transport.close()
+                    self.request.version, keep_alive, Config.REQUEST_TIMEOUT))
+            if keep_alive:
+                self.parser = None
+                self.request = None
+                self.url = None
+                self.headers = {}
+                self._total_request_size = 0
             else:
-                self.cleanup()
+                self.close()
         except Exception as e:
-            self.bail_out(
-                "Writing request failed, connection closed {}".format(e))
-
-    def bail_out(self, message):
-        log.error(message)
-        self.transport.close()
-
-    def cleanup(self):
-        self.parser = None
-        self.request = None
-        self.url = None
-        self.headers = None
-        self._total_request_size = 0
-
-    def close_if_idle(self):
-        """
-        Close the connection if a request is not being sent or received
-        :return: boolean - True if closed, false if staying open
-        """
-        if not self.parser:
-            self.transport.close()
-            return True
-        return False
+            log.error('Writing request failed, connection closed %s', e)
+            self.close()
 
 
 def serve(host, port, request_handler, after_start=None, before_stop=None,
```

Comment:
I kept the configuration settings as part of the arguments so other projects could re-purpose the server in their own frameworks without it being coupled with other parts of Sanic if they wanted.  But, that does come at the cost of passing lots of arguments.  What are your thoughts?